### DISCUSSION TITLE
AttributeError: 'unet' object has no attribute 'Conv'

### DISCUSSION THREAD

**Discussion by HILALOZTEMEL** [mlcommons/GaNDLF#427] [Q&A]
### > I think the error message is caused by the yaml file. My input image size is 128 x 128 . 
> 
 **> this is error mesage  :**

`(venv_gandlf) C:\Users\pc\GaNDLF>python gandlf_run  -c ./images_and_labels/model.yaml -i ./images_and_labels/train.csv -m ./images_and_labels/model -t True -d cpu
WARNING: Initializing 'norm_type' as 'batch'
WARNING: Initializing 'verbose' as False
WARNING: Initializing 'medcam_enabled' as False
WARNING: Initializing 'save_training' as False
WARNING: Initializing 'save_output' as False
WARNING: Initializing 'in_memory' as False
WARNING: Initializing 'pin_memory_dataloader' as False
WARNING: Initializing 'enable_padding' as False
WARNING: Initializing 'scaling_factor' as 1
WARNING: Initializing 'clip_grad' as None
WARNING: Initializing 'track_memory_usage' as False
WARNING: Initializing 'print_rgb_label_warning' as True
WARNING: Initializing 'data_postprocessing' as {}
WARNING: Initializing 'grid_aggregator_overlap' as crop
WARNING: Initializing 'clip_mode' as None
WARNING: Setting default step_size to: 0.02
Trying to resume training without changing any parameters from previous run.
Using previously saved parameter file ./images_and_labels/model\parameters.pkl
Using previously saved testing data ./images_and_labels/model\testing_0\data_testing.pkl
Using previously saved training+validation data ./images_and_labels/model\testing_0\data_trainAndVal.pkl
Number of channels :  1
Constructing queue for train data: 100%|████████████████████████| 643/643 [00:02<00:00, 216.35it/s]
C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torch\utils\data\dataloader.py:477: UserWarning: This DataLoader will create 16 worker processes in total. Our suggested max number of worker in current system is 4 (`cpuset` is not taken into account), which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  cpuset_checked))
Calculating weights
Constructing queue for penalty data: 100%|██████████████████████| 643/643 [00:02<00:00, 241.46it/s]
Looping over training data for penalty calculation: 100%|███████| 643/643 [00:05<00:00, 121.90it/s]
Constructing queue for validation data: 100%|███████████████████| 161/161 [00:01<00:00, 124.04it/s]
All Keys :  ['subject_id', '1', 'spacing', 'label', 'path_to_metadata']
Traceback (most recent call last):
  File "gandlf_run", line 116, in <module>
    args.reset,
  File "C:\Users\pc\GaNDLF\GANDLF\cli\main_run.py", line 91, in main_run
    reset=reset,
  File "C:\Users\pc\GaNDLF\GANDLF\training_manager.py", line 254, in TrainingManager
    testing_data=testingData,
  File "C:\Users\pc\GaNDLF\GANDLF\compute\training_loop.py", line 239, in training_loop
    ) = create_pytorch_objects(params, training_data, validation_data, device)
  File "C:\Users\pc\GaNDLF\GANDLF\compute\generic.py", line 69, in create_pytorch_objects
    model = get_model(parameters)
  File "C:\Users\pc\GaNDLF\GANDLF\models\__init__.py", line 106, in get_model
    return global_models_dict[params["model"]["architecture"]](parameters=params)
  File "C:\Users\pc\GaNDLF\GANDLF\models\unet.py", line 41, in __init__
    conv=self.Conv,
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torch\nn\modules\module.py", line 948, in __getattr__
    type(self).__name__, name))
AttributeError: 'unet' object has no attribute 'Conv'`

**> And this is my yaml file**

[model.pdf](https://github.com/CBICA/GaNDLF/files/8631955/model.pdf)

**Comment by sarthakpati** [comment]
Your issue is here:

![image](https://user-images.githubusercontent.com/11719673/166951592-3a8dfa21-9471-4c85-b323-a492f8831ab5.png)

GaNDLF currently only supports 2D and 3D computations.

**Reply by sarthakpati** [reply]
If you try with the latest master (i.e., after #429 was merged), you will see that there will be a run-time error invoked.

**Reply by HILALOZTEMEL** [reply]
> **yes it did, i didn't understand what to do in this situation**
`(venv_gandlf) C:\Users\pc\GaNDLF>python gandlf_run  -c ./images_and_labels/model.yaml -i ./images_and_labels/train.csv -m ./images_and_labels/model -t True -d cpu
WARNING: Initializing 'norm_type' as 'batch'
WARNING: Initializing 'verbose' as False
WARNING: Initializing 'medcam_enabled' as False
WARNING: Initializing 'save_training' as False
WARNING: Initializing 'save_output' as False
WARNING: Initializing 'in_memory' as False
WARNING: Initializing 'pin_memory_dataloader' as False
WARNING: Initializing 'enable_padding' as False
WARNING: Initializing 'scaling_factor' as 1
WARNING: Initializing 'clip_grad' as None
WARNING: Initializing 'track_memory_usage' as False
WARNING: Initializing 'print_rgb_label_warning' as True
WARNING: Initializing 'data_postprocessing' as {}
WARNING: Initializing 'grid_aggregator_overlap' as crop
WARNING: Initializing 'clip_mode' as None
WARNING: Setting default step_size to: 0.02
Trying to resume training without changing any parameters from previous run.
Number of channels :  1
Constructing queue for train data: 100%|██████████████████████████████████████████████████████████████████████| 643/643 [00:03<00:00, 195.17it/s]
C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torch\utils\data\dataloader.py:477: UserWarning: This DataLoader will create 16 worker processes in total. Our suggested max number of worker in current system is 4 (`cpuset` is not taken into account), which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  cpuset_checked))
Calculating weights
Constructing queue for penalty data: 100%|████████████████████████████████████████████████████████████████████| 643/643 [00:02<00:00, 219.22it/s]
Looping over training data for penalty calculation: 100%|█████████████████████████████████████████████████████| 643/643 [00:05<00:00, 113.44it/s]
Constructing queue for validation data: 100%|█████████████████████████████████████████████████████████████████| 161/161 [00:00<00:00, 170.19it/s]
All Keys :  ['subject_id', '1', 'spacing', 'label', 'path_to_metadata']
Since Device is CPU, Mixed Precision Training is set to False
Constructing queue for testing data: 100%|████████████████████████████████████████████████████████████████████| 201/201 [00:00<00:00, 208.51it/s]
Initializing training at : 2022/05/05::19:50:28
Using device: cpu
********************
********************
Starting Epoch :  0
********************
Starting Training :
********************
Looping over training data:   0%|                                                                                       | 0/6430 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "gandlf_run", line 116, in <module>
    args.reset,
  File "C:\Users\pc\GaNDLF\GANDLF\cli\main_run.py", line 91, in main_run
    reset=reset,
  File "C:\Users\pc\GaNDLF\GANDLF\training_manager.py", line 254, in TrainingManager
    testing_data=testingData,
  File "C:\Users\pc\GaNDLF\GANDLF\compute\training_loop.py", line 362, in training_loop
    model, train_dataloader, optimizer, params
  File "C:\Users\pc\GaNDLF\GANDLF\compute\training_loop.py", line 108, in train_network
    loss, calculated_metrics, _ = step(model, image, label, params)
  File "C:\Users\pc\GaNDLF\GANDLF\compute\step.py", line 81, in step
    output = model(image)
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torch\nn\modules\module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "C:\Users\pc\GaNDLF\GANDLF\models\unet.py", line 188, in forward
    x = self.de_3(x, x4)
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torch\nn\modules\module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "C:\Users\pc\GaNDLF\GANDLF\models\seg_modules\DecodingModule.py", line 46, in forward
    x = torch.cat([x1, x2], dim=1)
RuntimeError: Sizes of tensors must match except in dimension 1. Got 2 and 1 in dimension 4 (The offending index is 1)`

**Reply by sarthakpati** [reply]
Can you post the CSV and model config you are using?

**Reply by HILALOZTEMEL** [reply]
[train.csv](https://github.com/CBICA/GaNDLF/files/8634542/train.csv)
[model.zip](https://github.com/CBICA/GaNDLF/files/8634544/model.zip)

**Reply by sarthakpati** [reply]
Are your images RGB? If yes, you would need to set `n_channels: 3` under the `model` key.

**Reply by HILALOZTEMEL** [reply]
(venv_gandlf) C:\Users\pc\GaNDLF>python gandlf_run  -c ./images_and_labels/model.yaml -i ./images_and_labels/train.csv -m ./images_and_labels/model -t True -d cpu
Traceback (most recent call last):
  File "gandlf_run", line 116, in <module>
    args.reset,
  File "C:\Users\pc\GaNDLF\GANDLF\cli\main_run.py", line 29, in main_run
    parameters = parseConfig(model_parameters)
  File "C:\Users\pc\GaNDLF\GANDLF\parseConfig.py", line 112, in parseConfig
    params = yaml.safe_load(f)
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\yaml\__init__.py", line 125, in safe_load
    return load(stream, SafeLoader)
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\yaml\__init__.py", line 81, in load
    return loader.get_single_data()
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\yaml\constructor.py", line 49, in get_single_data
    node = self.get_single_node()
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\yaml\composer.py", line 36, in get_single_node
    document = self.compose_document()
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\yaml\composer.py", line 55, in compose_document
    node = self.compose_node(None, None)
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\yaml\composer.py", line 84, in compose_node
    node = self.compose_mapping_node(anchor)
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\yaml\composer.py", line 127, in compose_mapping_node
    while not self.check_event(MappingEndEvent):
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\yaml\parser.py", line 98, in check_event
    self.current_event = self.state()
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\yaml\parser.py", line 428, in parse_block_mapping_key
    if self.check_token(KeyToken):
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\yaml\scanner.py", line 116, in check_token
    self.fetch_more_tokens()
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\yaml\scanner.py", line 223, in fetch_more_tokens
    return self.fetch_value()
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\yaml\scanner.py", line 579, in fetch_value
    self.get_mark())
yaml.scanner.ScannerError: mapping values are not allowed here
  in "./images_and_labels/model.yaml", line 66, column 44

(venv_gandlf) C:\Users\pc\GaNDLF>'normalize': performs z-score
''normalize':' is not recognized as an internal or external command,
operable program or batch file.

(venv_gandlf) C:\Users\pc\GaNDLF>
(venv_gandlf) C:\Users\pc\GaNDLF>python gandlf_run  -c ./images_and_labels/model.yaml -i ./images_and_labels/train.csv -m ./images_and_labels/model -t True -d cpu
WARNING: Initializing 'norm_type' as 'batch'
WARNING: Initializing 'verbose' as False
WARNING: Initializing 'medcam_enabled' as False
WARNING: Initializing 'save_training' as False
WARNING: Initializing 'save_output' as False
WARNING: Initializing 'in_memory' as False
WARNING: Initializing 'pin_memory_dataloader' as False
WARNING: Initializing 'enable_padding' as False
WARNING: Initializing 'scaling_factor' as 1
WARNING: Initializing 'clip_grad' as None
WARNING: Initializing 'track_memory_usage' as False
WARNING: Initializing 'print_rgb_label_warning' as True
WARNING: Initializing 'data_postprocessing' as {}
WARNING: Initializing 'grid_aggregator_overlap' as crop
WARNING: Initializing 'clip_mode' as None
WARNING: Setting default step_size to: 0.02
Trying to resume training without changing any parameters from previous run.
Number of channels :  1
Constructing queue for train data: 100%|████████████████████████| 643/643 [00:03<00:00, 193.26it/s]
C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torch\utils\data\dataloader.py:477: UserWarning: This DataLoader will create 16 worker processes in total. Our suggested max number of worker in current system is 4 (`cpuset` is not taken into account), which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  cpuset_checked))
Calculating weights
Constructing queue for penalty data: 100%|██████████████████████| 643/643 [00:02<00:00, 232.13it/s]
Looping over training data for penalty calculation: 100%|███████| 643/643 [00:05<00:00, 120.66it/s]
Constructing queue for validation data: 100%|███████████████████| 161/161 [00:00<00:00, 186.98it/s]
All Keys :  ['subject_id', '1', 'spacing', 'label', 'path_to_metadata']
Since Device is CPU, Mixed Precision Training is set to False
Constructing queue for testing data: 100%|██████████████████████| 201/201 [00:00<00:00, 255.40it/s]
Initializing training at : 2022/05/06::13:22:17
Using device: cpu
********************
********************
Starting Epoch :  0
********************
Starting Training :
********************
Looping over training data:   0%|                                         | 0/3215 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "gandlf_run", line 116, in <module>
    args.reset,
  File "C:\Users\pc\GaNDLF\GANDLF\cli\main_run.py", line 91, in main_run
    reset=reset,
  File "C:\Users\pc\GaNDLF\GANDLF\training_manager.py", line 254, in TrainingManager
    testing_data=testingData,
  File "C:\Users\pc\GaNDLF\GANDLF\compute\training_loop.py", line 362, in training_loop
    model, train_dataloader, optimizer, params
  File "C:\Users\pc\GaNDLF\GANDLF\compute\training_loop.py", line 76, in train_network
    tqdm(train_dataloader, desc="Looping over training data")
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\tqdm\std.py", line 1195, in __iter__
    for obj in iterable:
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torch\utils\data\dataloader.py", line 517, in __next__
    data = self._next_data()
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torch\utils\data\dataloader.py", line 557, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torch\utils\data\_utils\fetch.py", line 44, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torch\utils\data\_utils\fetch.py", line 44, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torchio\data\queue.py", line 165, in __getitem__
    self._fill()
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torchio\data\queue.py", line 230, in _fill
    iterable = self.sampler(subject)
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torchio\data\sampler\sampler.py", line 89, in __call__
    raise RuntimeError(message)
**RuntimeError: Patch size (128, 128, 128) cannot be larger than image size (128, 128, 1)**

**Reply by HILALOZTEMEL** [reply]
**patch_size (128,128,1)
If I do, would it be wrong?**

**Reply by HILALOZTEMEL** [reply]
[model (2).zip](https://github.com/CBICA/GaNDLF/files/8639368/model.2.zip)

**Reply by HILALOZTEMEL** [reply]
**when i do this  `patch_size(128,128,1)` ,  i am getting this error**

![image](https://user-images.githubusercontent.com/60732734/167132964-a6ae140e-3970-4d0c-87f9-386c8785d09b.png)

**Reply by sarthakpati** [reply]
Can you remove `crop_external_zero_plane` from the preprocessing dict and try again?

**Reply by HILALOZTEMEL** [reply]
![image](https://user-images.githubusercontent.com/60732734/167145246-60e24257-2a13-44d2-9abb-f6776ecb244b.png)

**Reply by HILALOZTEMEL** [reply]
[model (3).zip](https://github.com/CBICA/GaNDLF/files/8641158/model.3.zip)

**Reply by sarthakpati** [reply]
Upon looking at your yaml, I see an issue that I have previously replied to [[ref](https://github.com/CBICA/GaNDLF/discussions/427#discussioncomment-2696484)]:

> Are your images RGB? If yes, you would need to set `n_channels: 3` under the `model` key.

**Reply by HILALOZTEMEL** [reply]
should it be `patch_size: [128,128,3]` when  `n_channels: 3`  ? @sarthakpati

**Reply by sarthakpati** [reply]
Nope, just `[128,128]` and `n_channels: 3` should suffice.

**Reply by HILALOZTEMEL** [reply]
![image](https://user-images.githubusercontent.com/60732734/167201555-ab50068c-3895-4913-95dc-72f75e4eaf5f.png)
[model (4).zip](https://github.com/CBICA/GaNDLF/files/8643103/model.4.zip)
i got an error like this

**Reply by sarthakpati** [reply]
Can you try again after removing the previous outputs? It seems like the number of channels are not getting picked up correctly.

**Reply by HILALOZTEMEL** [reply]
![image](https://user-images.githubusercontent.com/60732734/167206139-0f8ec256-48fd-48e3-bc22-9ed720bbdc56.png)
[model (5).zip](https://github.com/CBICA/GaNDLF/files/8643230/model.5.zip)

**Reply by sarthakpati** [reply]
The config does not have `n_channels: 3`.

**Reply by HILALOZTEMEL** [reply]
RuntimeError: Given groups=1, weight of size [30, 3, 3, 3, 3], expected input[1, 1, 128, 128, 1] to have 3 channels, but got 1 channels instead

**Reply by HILALOZTEMEL** [reply]
i get this error when ` n_channels: 3`

**Reply by sarthakpati** [reply]
Have you tried removing the previous output files before trying again?

**Reply by HILALOZTEMEL** [reply]
yes but it's the same

**Reply by sarthakpati** [reply]
Will you be able to send over an example case with annotation, along with the latest config file you are using?

**Reply by HILALOZTEMEL** [reply]
if i understood correctly what you said , 

**This is my label example** 

![image](https://user-images.githubusercontent.com/60732734/168376227-ec1358f5-617a-4943-9822-0dc5201abf96.png)

**and this is my images**

![image](https://user-images.githubusercontent.com/60732734/168376316-c55cb2c2-dfa8-4e3a-a6ef-20bb96cfbd56.png)

**and my yaml file**



[model (6).zip](https://github.com/CBICA/GaNDLF/files/8690796/model.6.zip)

**and my Error Message** 

(venv_gandlf) C:\Users\pc\GaNDLF>python gandlf_run  -c ./images_and_labels/model.yaml -i ./images_and_labels/train.csv -m ./images_and_labels/model -t True -d cpu
WARNING: Initializing 'norm_type' as 'batch'
WARNING: Initializing 'verbose' as False
WARNING: Initializing 'medcam_enabled' as False
WARNING: Initializing 'save_training' as False
WARNING: Initializing 'save_output' as False
WARNING: Initializing 'in_memory' as False
WARNING: Initializing 'pin_memory_dataloader' as False
WARNING: Initializing 'enable_padding' as False
WARNING: Initializing 'scaling_factor' as 1
WARNING: Initializing 'clip_grad' as None
WARNING: Initializing 'track_memory_usage' as False
WARNING: Initializing 'print_rgb_label_warning' as True
WARNING: Initializing 'data_postprocessing' as {}
WARNING: Initializing 'grid_aggregator_overlap' as crop
WARNING: Initializing 'clip_mode' as None
WARNING: Setting default step_size to: 0.02
Trying to resume training without changing any parameters from previous run.
Using previously saved parameter file ./images_and_labels/model\parameters.pkl
Using previously saved testing data ./images_and_labels/model\testing_0\data_testing.pkl
Using previously saved training+validation data ./images_and_labels/model\testing_0\data_trainAndVal.pkl
Number of channels :  3
Constructing queue for train data: 100%|█████████████████████████| 643/643 [00:09<00:00, 66.32it/s]
C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torch\utils\data\dataloader.py:477: UserWarning: This DataLoader will create 16 worker processes in total. Our suggested max number of worker in current system is 4 (`cpuset` is not taken into account), which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  cpuset_checked))
Calculating weights
Constructing queue for penalty data: 100%|██████████████████████| 643/643 [00:04<00:00, 150.52it/s]
Looping over training data for penalty calculation: 100%|████████| 643/643 [00:06<00:00, 96.77it/s]
Constructing queue for validation data: 100%|████████████████████| 161/161 [00:06<00:00, 25.08it/s]
All Keys :  ['subject_id', '1', 'spacing', 'label', 'path_to_metadata']
Since Device is CPU, Mixed Precision Training is set to False
Constructing queue for testing data: 100%|██████████████████████| 201/201 [00:01<00:00, 155.69it/s]
Initializing training at : 2022/05/13::22:18:58
Using device: cpu
********************
********************
Starting Epoch :  0
********************
Starting Training :
********************
Looping over training data:   0%|                                         | 0/6430 [00:12<?, ?it/s]
Traceback (most recent call last):
  File "gandlf_run", line 116, in <module>
    args.reset,
  File "C:\Users\pc\GaNDLF\GANDLF\cli\main_run.py", line 91, in main_run
    reset=reset,
  File "C:\Users\pc\GaNDLF\GANDLF\training_manager.py", line 254, in TrainingManager
    testing_data=testingData,
  File "C:\Users\pc\GaNDLF\GANDLF\compute\training_loop.py", line 362, in training_loop
    model, train_dataloader, optimizer, params
  File "C:\Users\pc\GaNDLF\GANDLF\compute\training_loop.py", line 108, in train_network
    loss, calculated_metrics, _ = step(model, image, label, params)
  File "C:\Users\pc\GaNDLF\GANDLF\compute\step.py", line 81, in step
    output = model(image)
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torch\nn\modules\module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "C:\Users\pc\GaNDLF\GANDLF\models\unet.py", line 177, in forward
    x1 = self.ins(x)
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torch\nn\modules\module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "C:\Users\pc\GaNDLF\GANDLF\models\seg_modules\in_conv.py", line 72, in forward
    x = self.conv0(x)
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torch\nn\modules\module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torch\nn\modules\conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "C:\Users\pc\anaconda3\envs\venv_gandlf\lib\site-packages\torch\nn\modules\conv.py", line 396, in _conv_forward
    self.padding, self.dilation, self.groups)
**RuntimeError: Given groups=1, weight of size [30, 3, 3, 3], expected input[1, 1, 128, 128] to have 3 channels, but got 1 channels instead**

**Reply by sarthakpati** [reply]
Hi, 

Can you please upload it to a cloud storage provider and send the link?

**Reply by HILALOZTEMEL** [reply]
https://drive.google.com/drive/u/0/folders/1pCLCocVfwHsyYDnUs4jLfT2TnfA9caCG

**Reply by sarthakpati** [reply]
The following model config works as expected for your data: [model.txt](https://github.com/CBICA/GaNDLF/files/8708143/model.txt)

Result shown for a small set for a single epoch:
![image](https://user-images.githubusercontent.com/11719673/168809847-449fea2a-7f1a-4c5f-9930-7f71e7bb68bd.png)

**Reply by HILALOZTEMEL** [reply]
i tried but i still get the same error

**Reply by sarthakpati** [reply]
Not sure what else I can do, since I have provided the model on which your data started training for me. I would suggest take a look at the model config I sent and ensure all previous results have been removed.

**Reply by HILALOZTEMEL** [reply]
i did the training but output_validation folder is empty ,ı can't solve the reason . What could be the reason?

**Reply by sarthakpati** [reply]
Unsure what you mean when you say "`output_validation` is empty". Are you trying to save the output validation masks for every epoch?

**Reply by HILALOZTEMEL** [reply]
yes , how can i do this?

**Reply by sarthakpati** [reply]
I am not sure. What do the training/validation logs show?
